sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/squid/CxxSquidSensor.java

```diff
@@ -175,99 +153,111 @@ private CxxConfiguration createConfiguration(FileSystem fs, Settings settings) {
     return cxxConf;
   }
 
-  private void save(Collection<SourceCode> squidSourceFiles) {
+  private void save(Collection<SourceCode> squidSourceFiles, SensorContext context) {
     int violationsCount = 0;
-    DependencyAnalyzer dependencyAnalyzer = new DependencyAnalyzer(resourcePerspectives, project, context, rules);
+    DependencyAnalyzer dependencyAnalyzer = new DependencyAnalyzer(context, rules);
 
     for (SourceCode squidSourceFile : squidSourceFiles) {
       SourceFile squidFile = (SourceFile) squidSourceFile;
       File ioFile = new File(squidFile.getKey());
-      InputFile inputFile = fs.inputFile(fs.predicates().is(ioFile));
+      InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates().is(ioFile));
 
-      saveMeasures(inputFile, squidFile);
-      saveFunctionAndClassComplexityDistribution(inputFile, squidFile);
-      saveFilesComplexityDistribution(inputFile, squidFile);
-      violationsCount += saveViolations(inputFile, squidFile);
-      //### @todo dependencyAnalyzer.addFile(inputFile, CxxParser.getIncludedFiles(ioFile));
```

Comment:
no need, ive removed the stuff that is not available anymore in server side. if this works it should actually be a nice feature. since it should detect circular dependencies and report those as issues. the duplicated include is at least working.